File: example_license_creation.ts

// Create and attach license terms for agent IP
const licenseTerms = {
  transferable: true,
  royaltyPolicy: "LAP", // License Agreement Policy
  commercialUse: true,
  commercialAttribution: true,
  derivativesAllowed: true,
  derivativesAttribution: true,
  derivativesApproval: false,
  derivativesReciprocal: true,
  currency: "0x...", // Payment token address
  uri: "ipfs://license-terms-metadata"
};

const attachResult = await client.license.attachTerms({
  ipId: agentIpId,
  licenseTerms: licenseTerms
});

File: example_license_creation.py

# Create and attach license terms using Python
license_terms = {
    "transferable": True,
    "royalty_policy": "LAP",
    "commercial_use": True,
    "commercial_attribution": True,
    "derivatives_allowed": True,
    "derivatives_attribution": True,
    "derivatives_approval": False,
    "derivatives_reciprocal": True,
    "currency": "0x...",  # Payment token address
    "uri": "ipfs://license-terms-metadata"
}

attach_result = client.license.attach_terms(
    ip_id=agent_ip_id,
    license_terms=license_terms
)
